File: backend/app/infrastructure/models/order.go
package models

import (
	"time"

	"github.com/uptrace/bun"
)

// OrderGroupModel 注文グループのインフラストラクチャーモデル
type OrderGroupModel struct {
	bun.BaseModel `bun:"table:order_groups"`

	OrdersID       string    `bun:"orders_id,pk"`
	TableSessionID string    `bun:"table_session_id,notnull"`
	Status         string    `bun:"status,notnull,default:'open'"`
	CreatedAt      time.Time `bun:"created_at,notnull,default:current_timestamp"`
}

// OrderItemModel 注文アイテムのインフラストラクチャーモデル
type OrderItemModel struct {
	bun.BaseModel `bun:"table:order_items"`

	OrderItemID  string    `bun:"order_item_id,pk"`
	OrdersID     string    `bun:"orders_id,notnull"`
	MenuID       string    `bun:"menu_id,notnull"`
	Quantity     int       `bun:"quantity,notnull"`
	PriceAtOrder float64   `bun:"price_at_order,notnull"`
	Status       string    `bun:"status,notnull,default:'pending'"`
	CreatedAt    time.Time `bun:"created_at,notnull,default:current_timestamp"`
}

File: frontend/apps/admin/app/(authenticated)/bill/types/order.ts
/** 注文アイテムオプション */
export type OrderItemOption = {
  menu_option_id: string;
};

/** 注文アイテム */
export type OrderItem = {
  order_item_id: string;
  menu_id: string;
  quantity: number;
  price_at_order: number;
  status: "pending" | "preparing" | "served" | "cancelled";
  created_at: string;
  options: OrderItemOption[];
};

/** 注文グループ */
export type OrderGroup = {
  orders_id: string;
  table_session_id: string;
  status: "open" | "closed" | "cancelled";
  created_at: string;
  items: OrderItem[];
};

/** GET /api/orders レスポンス */
export type GetOrdersResponse = {
  order_groups: OrderGroup[];
};
